packages/near-membrane-base/src/membrane.ts

```diff
@@ -937,29 +943,40 @@ export function createMembraneMarshall(
     }
 
     function serializeTarget(target: ProxyTarget): SerializedValue | undefined {
-        if (!ReflectHas(target, TO_STRING_TAG_SYMBOL)) {
-            // Fast path.
-            const brand = ReflectApply(ObjectProtoToString, target, []);
-            switch (brand) {
-                // The brand(s) below represent boxed primitives of `ESGlobalKeys`
-                // in packages/near-membrane-base/src/intrinsics.ts which are not
-                // remapped or reflective.
-                case '[object BigInt]':
-                    return SUPPORTS_BIG_INT ? serializeBooleanObject(target as any) : undefined;
-                case '[object Boolean]':
-                    return serializeBooleanObject(target as any);
-                case '[object Number]':
-                    return serializeNumberObject(target as any);
-                case '[object RegExp]':
-                    return serializeRegExp(target as any);
-                case '[object String]':
-                    return serializeStringObject(target as any);
-                case '[object Symbol]':
-                    return serializeSymbolObject(target as any);
-                default:
-                    return undefined;
-            }
+        try {
+            return ReflectHas(target, TO_STRING_TAG_SYMBOL)
+                ? serializeTargetByTrialAndError(target)
```

Comment:
omg I love the concept and the name

Reply:
It's a technique from lodash. Take large complex functions that have forks for fast paths and make the initial function a callout to just the fast and slow paths. this allows for optimizing fast paths that would otherwise bust the inline caches when nested as 1 large function.